Project_Mars/Project_Mars/Pages/LanguageTab.cs

```diff
@@ -0,0 +1,270 @@
+
+            if (totalrows > 0)
+            {
+
+                Wait.WaitToBeVisible(driver, "XPath", "//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table//tbody", 4);
+
+                int countrows = driver.FindElements(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table//tbody")).Count;
+                for (int i = 1; i <= countrows; i++)
+                {
+                    Thread.Sleep(2000);
+                    IWebElement selectLanguageToUpdate = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[" + i + "]/tr/td[1]"));
+
+                    if (selectLanguageToUpdate.Text.Equals(OldName))
+                    {
+                        Thread.Sleep(2000);
+                        IWebElement editLanguageButton = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[" + i + "]/tr/td[3]/span[1]/i"));
+                        Thread.Sleep(1000);
+                        editLanguageButton.Click();
+                        Thread.Sleep(1000);
+                        IWebElement editLanguageTextbox = driver.FindElement(By.XPath(" //*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[" + i + "]/tr/td/div/div[1]/input"));
+                        editLanguageTextbox.Click();
+                        editLanguageTextbox.Clear();
+                        editLanguageTextbox.SendKeys(NewName);
+                        Thread.Sleep(2000);
+                        IWebElement editLevelDropdown = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[" + i + "]/tr/td/div/div[2]/select"));
+                        editLevelDropdown.Click();
+                        if (NewLevel.Equals("Basic"))
+                        {
+                            IWebElement levelOption = driver.FindElement(By.XPath("//option[@value='Basic']"));
+                            levelOption.Click();
+                        }
+                        else if (NewLevel.Equals("Conversational"))
+                        {
+                            IWebElement levelOption = driver.FindElement(By.XPath("//option[@value='Conversational']"));
+                            levelOption.Click();
+                        }
+                        else if (NewLevel.Equals("Fluent"))
+                        {
+                            IWebElement levelOption = driver.FindElement(By.XPath("//option[@value='Fluent']"));
+                            levelOption.Click();
+                        }
+                        else if (NewLevel.Equals("Native/Bilingual"))
```

Comment:
also rather then doing a if block you could do something like   string skillLevelXpath = "//option[contains(text(),'" + skillLevel + "')]";

Reply:
and then you have the xpath which you can use your find by element

Reply:
Added a function 
![image](https://github.com/Shilpii01/Project_Mars/assets/145233624/9f279dfe-751d-4ca7-aeb7-787da083d983)